lib/graphql/internal_representation/rewrite.rb

```diff
@@ -138,6 +144,12 @@ def validate(context)
           # Pop this field's node
           # and record any directives that were visited
           # during this field & before it (eg, inline fragments)
+          
+          return if ast_node.directives.any? do |directive|
+            directive.name == 'skip' && directive.arguments.any? {|arg| arg.value == true } ||
+            directive.name == 'include' && directive.arguments.any? {|arg| arg.value == false }
+          end
```

Comment:
And this is a corresponding change for the one above, right? Since we shortcutted the `enter` hook, we should also shortcut the `leave` hook?